### PR TITLE
Update amqp10_client and amqp10_common for hex.pm publish

### LINKED ISSUE #2577: rabbitmq-amqp1.0-common: publish to hex.pm
This library should be published on hex.pm in order to publish `amqp10_client` to hex.pm as well.
At the moment using `amqp10_client` is a quite painful process.

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #2577

Related rabbitmq/rabbitmq-ci#50

### CODE REVIEW

File: deps/amqp10_client/Makefile [hunk 1]
```diff
@@ -2,13 +2,30 @@ PROJECT = amqp10_client
 PROJECT_DESCRIPTION = AMQP 1.0 client from the RabbitMQ Project
 PROJECT_MOD = amqp10_client_app
 
+define PROJECT_APP_EXTRA_KEYS
+%% Hex.pm package informations.
+	{licenses, ["MPL 2.0"]},
```
Comment (dumbbell) [inline deps/amqp10_client/Makefile:7]: Hex.pm now asks for an SPDX-compatible license short name. It should be `MPL-2.0` in this case.

Comment (lukebakken) [reply deps/amqp10_client/Makefile]: Good catch! Got it in the `amqp_client` `Makefile` as well.

**Comment by lukebakken** [comment]
@pjk25 @dumbbell OK I think this is ready for a final review. Here's how I tested these changes:

* Published to hex.pm:
    ```
    cd deps/amqp10_common
    make RABBITMQ_VERSION=3.10.0-rc.1
    make RABBITMQ_VERSION=3.10.0-rc.1 hex-publish

    cd ../amqp10_client
    make RABBITMQ_VERSION=3.10.0-rc.1
    make RABBITMQ_VERSION=3.10.0-rc.1 hex-publish
    ```

    Produces:
    https://hex.pm/packages/amqp10_client
    https://hex.pm/packages/amqp10_common

* Created test app - https://github.com/lukebakken/amqp10_test
* Cloning and compiling test app works:
    ```
    lbakken@shostakovich ~/development/lukebakken/amqp10_test (main=)
    $ rebar3 compile
    ===> Verifying dependencies...
    ===> Fetching amqp10_client v3.10.0-rc.1
    ===> Fetching amqp10_common v3.10.0-rc.1
    ===> Analyzing applications...
    ===> Compiling amqp10_common
    ===> Compiling amqp10_client
    ===> Analyzing applications...
    ===> Compiling amqp10_test
    ```

**Review by pjk25 — APPROVED** [review]
Code looks good and at least one user has tested the package according to #2577

**Pull request merged**